src/libcollections/string.rs

```diff
@@ -40,6 +40,7 @@ pub struct String {
 }
 
 /// A possible error value from the `String::from_utf8` function.
+#[deriving(Show)]
```

Comment:
This may not want to necessarily derive `Show` as dumping a huge amount of bytes to output doesn't seem super useful, could this just delegate to `String`?